src/main/java/org/testobject/test/LogentriesResultReporter.java

```diff
@@ -0,0 +1,69 @@
+package org.testobject.test;
+
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+import java.util.HashMap;
+import java.util.Map;
+
+public class LogentriesResultReporter extends TestWatcher {
+	private static final String LOGENTRIES_URL = "http://webhook.logentries.com/noformat/logs/";
+
+	private static final String JSON_TEST_FIELD = "test";
+	private static final String JSON_DURATION_FIELD = "duration";
+
+	private final Client client = ClientBuilder.newBuilder().build();
+
+	private final WebTarget webTarget;
+
+	private final Map<String, Long> testStartTimes = new HashMap<>();
+
+	public LogentriesResultReporter(String logentriesToken) {
+		this.webTarget = client.target(LOGENTRIES_URL).path(logentriesToken);
+	}
+
+	public LogentriesResultReporter() {
+		String logentriesToken = System.getenv("LOGENTRIES_TOKEN");
+		if (logentriesToken == null) {
+			throw new IllegalStateException("LOGENTRIES_TOKEN env was empthy. Please set the env variable");
+		}
+		this.webTarget = client.target(LOGENTRIES_URL).path(logentriesToken);
```

Comment:
What about calling `this(logentriesToken)`? I think it would reduce code duplication and make it easier to change the constructor.